### ISSUE TITLE
PR builds failing due to "ELF section name out of range"

### ISSUE THREAD

**Issue by pjaitken** [fluent/fluent-bit#1730]
## Bug Report

**Describe the bug**
Multiple pull request builds are failing with "ELF section name out of range". This seems unrelated to the PR diffs.

Some examples:

Build|Error
--|--
https://travis-ci.org/fluent/fluent-bit/jobs/608054594 | /usr/bin/ld.gold: fatal error: ../../library/libmonkey.a: ELF section name out of range
https://travis-ci.org/fluent/fluent-bit/jobs/608054598 | /usr/bin/ld.gold: fatal error: ../../library/libmbedcrypto.a: ELF section name out of range
https://travis-ci.org/fluent/fluent-bit/jobs/609133611 | /usr/bin/ld.gold: fatal error: ../../library/libflb-plugin-filter_kubernetes.a: ELF section name out of range
https://travis-ci.org/fluent/fluent-bit/jobs/608866676 | /usr/bin/ld.gold: fatal error: ../../library/librbtree.a: ELF section name out of range
https://travis-ci.org/fluent/fluent-bit/jobs/610869822 | /usr/bin/ld.gold: fatal error: ../../lib/libluajit.a: ELF section name out of range

**To Reproduce**
* submit PR

**Expected behavior**
* Builds don't fail.

**Comment by fujimotos** [comment]
> Multiple pull request builds are failing with "ELF section name out of range". This seems unrelated to the PR diffs.

I took a look at the build log this morning and it occurred to me that those
failures are occurring because we specifically use `gold` (instead of `ld`)
as a linker when sanitizers enabled.

I posted a sentinel PR to see if the hypthesis holds up (#1738)

> https://travis-ci.org/fluent/fluent-bit/builds/611697896

... and indeed it seems to fix the issue, without breaking anything.

So I think this issue is going to be resolved when #1738 is merged.
